common/djangoapps/third_party_auth/tests/test_pipeline_integration.py

```diff
@@ -336,3 +334,57 @@ def test_provider_settings_redirect_to_registration(self, send_to_registration_f
                 )
                 assert response.status_code == 302
                 assert response.url == expected_redirect_url
+
+
+@unittest.skipUnless(testutil.AUTH_FEATURE_ENABLED, testutil.AUTH_FEATURES_KEY + ' not enabled')
+class UserDetailsForceSyncTestCase(testutil.TestCase, test.TestCase):
+    """Tests to ensure learner profile data is properly synced if the provider requires it."""
+
+    def setUp(self):
+        super(UserDetailsForceSyncTestCase, self).setUp()
+        self.user = UserFactory.create()  # pylint: disable=attribute-defined-outside-init
+        self.details = {
+            'email': 'new+{}'.format(self.user.email),
+            'username': 'new_{}'.format(self.user.username),
+            'fullname': 'Grown Up {}'.format(self.user.profile.name),
+            'country': 'PK',
+            'non_existing_field': 'value',
+        }
+
+        # Mocks
+        self.strategy = mock.MagicMock()
+
+        get_from_pipeline = mock.patch('third_party_auth.pipeline.provider.Registry.get_from_pipeline')
+        self.get_from_pipeline = get_from_pipeline.start()
+        self.get_from_pipeline.return_value = mock.MagicMock(sync_learner_profile_data=True)
+        self.addCleanup(get_from_pipeline.stop)
+
+        send_mail = mock.patch('third_party_auth.pipeline.send_mail')
+        self.send_mail = send_mail.start()
+        self.addCleanup(send_mail.stop)
+
+    def test_user_details_force_sync(self):
+        """
+        The user details are synced properly and an email is sent when the email is changed.
+        """
+        # Keep old information for reference
+        old_email = self.user.email
+        old_username = self.user.username
+        old_fullname = self.user.profile.name
+
+        # Begin the pipeline.
+        pipeline.user_details_force_sync(
+            auth_entry=pipeline.AUTH_ENTRY_LOGIN,
+            strategy=self.strategy,
+            details=self.details,
+            user=self.user,
+        )
+
+        # User now has updated information.
+        assert self.user.email == 'new+{}'.format(old_email)
+        assert self.user.username == 'new_{}'.format(old_username)
+        assert self.user.profile.name == 'Grown Up {}'.format(old_fullname)
+        assert self.user.profile.country == 'PK'
+
+        # Mail was sent properly.
+        self.send_mail.assert_called_once()
```

Comment:
Oh, I forgot: I need to modify this to show what it was called with, for completeness.